blocks/model.py

```diff
@@ -190,7 +179,13 @@ def get_params(self):
         parameters that do not belong to any brick.
 
         """
-        return self.params
+        if (len(set(parameter.name for parameter in self.parameters)) !=
+                len(self.parameters)):
+            raise ValueError("parameters should have different names")
```

Comment:
Yes, you should use hierarchical names.

No, you can get the brick a parameter belongs to by exploring its tag. Then you can use the `get_unique_name` routine of this brick.